utils/s2n_set.c

```diff
@@ -54,61 +55,62 @@ static int s2n_set_binary_search(struct s2n_set *set, void *element, uint32_t* o
     }
 
     *out = low;
-    return S2N_SUCCESS;
+    return S2N_RESULT_OK;
 }
 
 struct s2n_set *s2n_set_new(size_t element_size, int (*comparator)(const void*, const void*))
 {
     notnull_check_ptr(comparator);
     struct s2n_blob mem = {0};
-    GUARD_PTR(s2n_alloc(&mem, sizeof(struct s2n_set)));
+    GUARD_POSIX_PTR(s2n_alloc(&mem, sizeof(struct s2n_set)));
     struct s2n_set *set = (void *) mem.data;
     *set = (struct s2n_set) {.data = s2n_array_new(element_size), .comparator = comparator};
     if(set->data == NULL) {
-        s2n_free(&mem);
+        GUARD_POSIX_PTR(s2n_free(&mem));
         return NULL;
     }
     return set;
 }
 
-int s2n_set_add(struct s2n_set *set, void *element)
+S2N_RESULT s2n_set_add(struct s2n_set *set, void *element)
 {
-    uint32_t index;
-    GUARD(s2n_set_binary_search(set, element, &index));
-    GUARD(s2n_array_insert_and_copy(set->data, element, index));
-    return S2N_SUCCESS;
+    uint32_t index = 0;
+    GUARD_RESULT(s2n_set_binary_search(set, element, &index));
+    GUARD_AS_RESULT(s2n_array_insert_and_copy(set->data, element, index));
+    return S2N_RESULT_OK;
 }
 
 void *s2n_set_get(struct s2n_set *set, uint32_t index)
 {
     return s2n_array_get(set->data, index);
 }
 
-int s2n_set_remove(struct s2n_set *set, uint32_t index)
+S2N_RESULT s2n_set_remove(struct s2n_set *set, uint32_t index)
 {
-    return s2n_array_remove(set->data, index);
+    GUARD_AS_RESULT(s2n_array_remove(set->data, index));
+    return S2N_RESULT_OK;
 }
-    
-int s2n_set_free_p(struct s2n_set **pset)
+
+S2N_RESULT s2n_set_free_p(struct s2n_set **pset)
 {
-    notnull_check(pset);
+    ENSURE_NONNULL(pset);
     struct s2n_set *set = *pset;
 
-    notnull_check(set);
-    GUARD(s2n_array_free(set->data));
-    GUARD(s2n_free_object((uint8_t **)pset, sizeof(struct s2n_set)));
+    ENSURE_NONNULL(set);
+    GUARD_AS_RESULT(s2n_array_free(set->data));
+    GUARD_AS_RESULT(s2n_free_object((uint8_t **)pset, sizeof(struct s2n_set)));
 
-    return S2N_SUCCESS;
+    return S2N_RESULT_OK;
 
 }
 
-int s2n_set_free(struct s2n_set *set)
+S2N_RESULT s2n_set_free(struct s2n_set *set)
 {
     return s2n_set_free_p(&set);
 }
 
 
-int s2n_set_size(struct s2n_set *set)
+uint32_t s2n_set_size(struct s2n_set *set)
```

Comment:
This is confusingly named.  Not worth fixing for this PR, but makes me thing you're "setting the size" to something :P

Reply:
haha it threw me off as well!